src/sagemaker/fw_utils.py

```diff
@@ -475,6 +484,33 @@ def empty_framework_version_warning(default_version, latest_version):
     return " ".join(msgs)
 
 
+def warn_if_parameter_server_with_multi_gpu(training_instance_type, distributions):
+    """Distributed training with parameter server and multi-GPU instances is not
+    supported. Warn the user that training will not fully leverage all the GPU
+    cores if parameter server is enabled and a multi-GPU instance is selected.
+
+    Args:
+        training_instance_type (str): A string representing the type of training instance selected.
+        distributions (dict): A dictionary with information to enable distributed training.
+                              (Defaults to None if distributed training is not enabled.)
+
+    """
+    multi_gpu_instance_types = {
```

Comment:
can we use black list instead of white list? if instance type is gpu and is not p2.xlarge or p3.2xlarge,...

Reply:
reason is new gpu instance can be introduced in the future, we do not want to update them all the time.

Reply:
is there a different way we could determine this?

I'm thinking something like

```python
instance_type = instance_type.split(".")
instance_type[1].startswith("p") and int(instance_type[2].split("x")[0]) % 8 == 0
```
though obviously it could be written a lot more cleanly (regex?) than what I just did...

Reply:
Sounds good, I've updated it to check against a single-GPU blacklist.